From 146fd82eaa5098c23cf191b818d49bcc90d3ab2f Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Tue, 4 Jul 2017 01:32:30 -0400
Subject: [PATCH 110/190] mark open_softirq as only used for init

[nicolas.bouchinet@ssi.gouv.fr: Adapt to commit 75e340ce106fa]
---
 include/linux/interrupt.h | 2 +-
 kernel/softirq.c          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/interrupt.h b/include/linux/interrupt.h
index 51b6484c0493..66ce30c94387 100644
--- a/include/linux/interrupt.h
+++ b/include/linux/interrupt.h
@@ -599,11 +599,11 @@ static inline void do_softirq_post_smp_call_flush(unsigned int unused)
 {
 	do_softirq();
 }
 #endif
 
-extern void open_softirq(int nr, void (*action)(void));
+extern void __init open_softirq(int nr, void (*action)(void));
 extern void softirq_init(void);
 extern void __raise_softirq_irqoff(unsigned int nr);
 
 extern void raise_softirq_irqoff(unsigned int nr);
 extern void raise_softirq(unsigned int nr);
diff --git a/kernel/softirq.c b/kernel/softirq.c
index 77198911b8dd..c83209f8458b 100644
--- a/kernel/softirq.c
+++ b/kernel/softirq.c
@@ -788,11 +788,11 @@ void __raise_softirq_irqoff(unsigned int nr)
 	lockdep_assert_irqs_disabled();
 	trace_softirq_raise(nr);
 	or_softirq_pending(1UL << nr);
 }
 
-void open_softirq(int nr, void (*action)(void))
+void __init open_softirq(int nr, void (*action)(void))
 {
 	softirq_vec[nr].action = action;
 }
 
 /*
-- 
2.52.0

